### PR TITLE
feat: stacks nft collectibles and nft metatdata

### LINKED ISSUE #3200: Move Stacks NFTs and names into "Collectibles" area with placeholder images
We should move the current listing of NFTs and names into the Collectibles area for listing after Ordinal inscriptions, displaying them with placeholder images there until later when we wire them up with https://github.com/hirosystems/stacks-wallet-web/issues/1028

https://www.figma.com/file/OSQocEEJnfcMkniEl8u3U8/%231483---Initial-Bitcoin-support?node-id=1182%3A76377&t=H4pGkYXsGMLRHOAN-1

Image

E.g. that if it currently looks like this:

Image

Then it should say "bitcoin-bears" where it's labeled "NFT name" in Figma.

Note how the placeholder image is different if its a BNS name vs. NFT.

If there are multiple of the same NFT, then we should show several duplicate blocks under collectibles for it, corresponding to the number.

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/4245579284).

This PR does two things:

1. Relocates bns names and nfts to be listed after ordinals under `Collectibles`
2. Adds a query to get nft metadata to show the nft image
3. Adds a kill switch into our wallet config so the feature can be turned off if needed

I have also refactored the `collectibles` code a bit and made responsive from 4 grid items to 1. I also added the hover state but w/out the text until I understand better what information we want displayed on nfts.

Screen Shot 2023-02-22 at 3 15 14 PM

Screen Shot 2023-02-22 at 3 15 41 PM

Screen Shot 2023-02-22 at 3 16 12 PM

### CODE REVIEW

File: src/app/features/collectibles/stacks-non-fungible-tokens.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import type { StacksNonFungibleTokenAssetBalance } from '@shared/models/crypto-asset-balance.model';
+
+import { useCurrentAccountDisplayName } from '@app/common/hooks/account/use-account-names';
+
+import { BaseCollectible } from './components/base-collectible';
+
+interface StacksNftCryptoAssetsProps {
+  assetBalances: StacksNonFungibleTokenAssetBalance[];
+}
+export function StacksNonFungibleTokens({ assetBalances }: StacksNftCryptoAssetsProps) {
```
Comment (kyranjamie) [inline src/app/features/collectibles/stacks-non-fungible-tokens.tsx]: Lots of duplicate code, layout component would be great + separate bns/nft components if they're that different.

Comment (fbwoolf) [reply src/app/features/collectibles/stacks-non-fungible-tokens.tsx]: Yeah, I agree but was trying to get out fast today and didn't want to step on Edu's toes if he was planning to come back and refactor the collectibles code? Seems like quite a bit is still commented out so wasn't sure. @edu-stx?

Comment (fbwoolf) [reply src/app/features/collectibles/stacks-non-fungible-tokens.tsx]: I had setup a `crypto-assets` folder under `components` where you can see I moved the list from. I had a `StacksNonFungibleTokenAssetItem` and a layout component setup as a placeholder for this work. But, was rolling with `Collectibles` now that Edu set it up.

**Comment by fbwoolf** [comment]
Added the nft metadata query here and did some refactoring @kyranjamie if want to review again.

File: src/app/features/collectibles/components/collectible.layout.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  children,
+}: CollectibleLayoutProps) {
+  return (
+    <Box
+      _focus={{
+        outline: '4px solid #CEDAFA',
+      }}
+      as={onClick ? 'button' : 'div'}
+      borderRadius="16px"
+      onClick={onClick}
```
Comment (kyranjamie) [inline src/app/features/collectibles/components/collectible.layout.tsx]: nit picking but a custom `onSelectCollectible` event is nicer/more readable imo

File: src/app/features/collectibles/components/stacks-non-fungible-tokens.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  const [bkgrdProps, setBkgrdProps] = useState(backgroundProps);
+  const isImageAvailable = metadata && metadata.cached_image && isValidUrl(metadata?.cached_image);
+  const placeholderImage = StacksNft;
+
+  const onImageError = (evt: SyntheticEvent<HTMLImageElement>) => {
+    evt.currentTarget.src = StacksNft;
+    evt.currentTarget.width = 100;
+    setBkgrdProps(placeholderBackgroundProps);
+  };
+
```
Comment (kyranjamie) [inline src/app/features/collectibles/components/stacks-non-fungible-tokens.tsx:35]: suggest early return pattern

```
if (!isImageAvailable) return
```

**Review by kyranjamie — APPROVED** [review]
Really clean tidy up Fara, great work

File: src/app/features/collectibles/components/collectible.layout.tsx (hunk 2 above)
Comment (fbwoolf) [reply src/app/features/collectibles/components/collectible.layout.tsx]: Agree here, missed changing it. Will do.

**Review by edu-stx — APPROVED** [review]
Brilliant

**Pull request merged**